IPlug/CLAP/IPlugCLAP.h

```diff
@@ -0,0 +1,250 @@
+ */
+
+#include "IPlugAPIBase.h"
+#include "IPlugProcessor.h"
+#include "plugin.hh"
+
+#include "config.h"   // This is your plugin's config.h.
+
+BEGIN_IPLUG_NAMESPACE
+
+/** Used to pass various instance info to the API class */
+struct InstanceInfo
+{
+  const clap_plugin_descriptor* mDesc;
+  const clap_host* mHost;
+};
+
+// Set the level of host checking based on if this is debug build
+
+#ifdef _DEBUG
+using ClapPluginHelper = clap::helpers::Plugin<clap::helpers::MisbehaviourHandler::Terminate, clap::helpers::CheckingLevel::Maximal>;
+using ClapHost = clap::helpers::HostProxy<clap::helpers::MisbehaviourHandler::Terminate, clap::helpers::CheckingLevel::Maximal>;
+#else
+using ClapPluginHelper = clap::helpers::Plugin<clap::helpers::MisbehaviourHandler::Ignore, clap::helpers::CheckingLevel::None>;
+using ClapHost = clap::helpers::HostProxy<clap::helpers::MisbehaviourHandler::Ignore, clap::helpers::CheckingLevel::None>;
+#endif
+
+/** CLAP API base class for an IPlug plug-in
+*   @ingroup APIClasses */
+class IPlugCLAP : public IPlugAPIBase
+                , public IPlugProcessor
+                , public ClapPluginHelper
+{
+  struct ParamToHost
+  {
+    enum class Type { Begin, Value, End };
+
+    ParamToHost()
+      : mType(Type::Value)
+      , mIdx(-1)
+      , mValue(0.0)
+    {}
+
+    ParamToHost(Type type, int idx, double value)
+      : mType(type)
+      , mIdx(idx)
+      , mValue(value)
+    {}
+
+    uint32_t idx() const { return static_cast<uint32_t>(mIdx); }
+    double value() const { return mValue; }
+    uint16_t type() const
+    {
+      switch (mType)
+      {
+        case Type::Begin:   return CLAP_EVENT_PARAM_GESTURE_BEGIN;
+        case Type::End:     return CLAP_EVENT_PARAM_GESTURE_END;
+        case Type::Value:   // fall through
+        default:            return CLAP_EVENT_PARAM_VALUE;
+      }
+    }
+    
+    Type mType;
+    int mIdx;
+    double mValue;
+  };
+  
+public:
+  IPlugCLAP(const InstanceInfo& info, const Config& config);
+
+  // IPlugAPIBase
+  void BeginInformHostOfParamChange(int idx) override;
+  void InformHostOfParamChange(int idx, double normalizedValue) override;
+  void EndInformHostOfParamChange(int idx) override;
+//  void InformHostOfPresetChange() override;
+  bool EditorResize(int viewWidth, int viewHeight) override;
+  
+  // IPlugProcessor
+  void SetTailSize(int tailSize) override;
+  void SetLatency(int samples) override;
+  bool SendMidiMsg(const IMidiMsg& msg) override;
+  bool SendSysEx(const ISysEx& msg) override;
+
+private:
+  
+  // clap_plugin
+  bool init() noexcept override;
+  bool activate(double sampleRate, uint32_t minFrameCount, uint32_t maxFrameCount) noexcept override;
+  void deactivate() noexcept override;
+  bool startProcessing() noexcept override { return true; }
+  void stopProcessing() noexcept override {}
+  clap_process_status process(const clap_process *process) noexcept override;
+  
+  //void onMainThread() noexcept override {}
```

Comment:
@AlexHarker Also here?

Reply:
This probably needs checking - the commented out things just mean there was a virtual in the base helper class that might need overriding, but I'm not sure what this method is supposed to do.

Reply:
Removed for now